python/run_look_with_filters.py

```diff
@@ -0,0 +1,75 @@
+import json
+import sys
+from typing import cast, Dict, List, Union
+
+from looker_sdk import client, models
+
+sdk = client.setup("../looker.ini")
+
+
+def main():
+    """Given a look id, obtain the query behind it and run it with the desired
+     filter values.
+    """
+    look_id = sys.argv[1] if len(sys.argv) > 1 else ""
+    filter_args = iter(sys.argv[2:])
+    filter_values: List[Dict[str, str]] = []
+
+    if not (look_id and len(sys.argv[2:]) > 0 and len(sys.argv[2:]) % 2 == 0):
+        print(
+            "Please provide: <lookId> <filter_1> <filter_value_1> "
+            "<filter_2> <filter_value_2> ..."
+        )
+        return
+
+    for filter_name in filter_args:
+        filter_value = next(filter_args)
+        filter_values.append({filter_name: filter_value})
+
+    query = get_look_query(int(look_id))
+    for filters in filter_values:
+        results = run_query_with_filter(query, filters)
```

Comment:
the code looks good to me (ditto on error handling - "look not found" etc).

the only confusing part (and this could certainly just be me as a looker noob) is that I was expecting a single query with all the filters applied but it seems like you're querying once per filter? I don't know which way it "should" be but either way it might be good to call that out in the `main()` docstring

Reply:
Good call on the error handling. I think I will create a module `exceptions.py` containing exceptions like `LookNotFound`, `DashboardNotFound`, `LookIdError`, etc. This will allow us to reuse the exceptions in other examples.

@joeldodge79 You could definitely run the query with all the filters in the where clause using the OR operator (if they're all referring to the same filter) for example. However, there are times where you'd want to run the Look with each filter value. This makes me realise that I can make example cleaner by providing the ability to pass a number of filter values for one query run and instead of looping we'd just be applying them once.